modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -2321,7 +2321,13 @@ public CodegenOperation fromOperation(String path,
                     }
 
                     // generate examples
-                    op.examples = new ExampleGenerator(schemas, openAPI).generateFromResponseSchema(responseSchema, getProducesInfo(openAPI, operation));
+                    String exampleStatusCode = "200";
+                    for (String key : operation.getResponses().keySet()) {
+                        if (operation.getResponses().get(key) == methodResponse && !key.equals("default")) {
```

Comment:
`methodResponse` is ["2xx" or "default"](https://github.com/ackintosh/openapi-generator-1/blob/2e4f484af9906aa78c0f7bd7cb7c470a0de66ac6/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java#L2278).